// carbon-components-react/lib/components/InlineCheckbox/InlineCheckbox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var InlineCheckbox = function InlineCheckbox(_ref) {
  var id = _ref.id,
      checked = _ref.checked,
      ariaLabel = _ref.ariaLabel,
      name = _ref.name,
      onClick = _ref.onClick,
      onKeyDown = _ref.onKeyDown;
  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("input", {
    id: id,
    className: "bx--checkbox",
    type: "checkbox",
    name: name,
    checked: checked,
    onClick: onClick,
    onKeyDown: onKeyDown
  }), _react.default.createElement("label", {
    htmlFor: id,
    className: "bx--checkbox-label",
    "aria-label": ariaLabel
  }));
};

InlineCheckbox.propTypes = {
  /**
   * Provide an `id` for the underlying input control
   */
  id: _propTypes.default.string.isRequired,

  /**
   * Provide a `name` for the underlying input control
   */
  name: _propTypes.default.string.isRequired,

  /**
   * Specify the label for the control
   */
  ariaLabel: _propTypes.default.string.isRequired,

  /**
   * Provide a handler that is invoked when a user clicks on the control
   */
  onClick: _propTypes.default.func,

  /**
   * Provide a handler that is invoked on the key down event for the control
   */
  onKeyDown: _propTypes.default.func,

  /**
   * Specify whether the underlying control is checked, or not
   */
  checked: _propTypes.default.bool.isRequired
};
var _default = InlineCheckbox;
exports.default = _default;
